src/main/java/com/fasterxml/jackson/databind/deser/BasicDeserializerFactory.java

```diff
@@ -361,7 +361,10 @@ private boolean _addExplicitDelegatingCreators(DeserializationContext ctxt,
         final AnnotationIntrospector intr = ctxt.getAnnotationIntrospector();
         boolean added = false;
 
-        for (PotentialCreator ctor : potentials) {
+        // [databind#4724] since 2.18.1 Let's iterate creators in reverse because collection of creators
```

Comment:
I am not sure I follow this comment... does _addExplicitDelegatingCreator override possible formerly added ones or... ?

Reply:
> does _addExplicitDelegatingCreator override possible formerly added ones

Yeah, via `BasicDeserializer._handleSingleArgumentCreator()` -> `CreatorCollector.addStringCreator()`.

Reply:
Hmmmh. I would have thought a conflict should arise if multiple explicit creators of same type exist. I'll need to think this through; I am not comfortable with change I don't understand. :)

Reply:
>  if multiple explicit creators of same type exist

This may be a problem as well. There seems to be no forcing deduplication.

Reply:
There is intended logic; but precedence only works within category... so it might not be considering choices as being similar enough to collide.

But the basic rule is that an explicitly annotated Creator does override (and not conflict) default one; both Record case and POJOs. Default one does similarly have precedence over implicit ones; implicit ones do not conflict (if multiple exist, none chosen).
But explicitly annotated constructors should conflict -- except there's Properties- vs Delegating which are not conflicting.

I'll need to check out this example again to see if I agree with intended resolution.

Reply:
> But the basic rule is that an explicitly annotated Creator does override (and not conflict) default one; both Record case and POJOs.

I read through your explanation few times more. I think we might enjoy introducing the concept of "default creators".

Currently the primary constructor of record class is being considered just as one of "explicitDelegatingConstructor". At the time of setting the creator to use, we can't use default-ness to choose the right creator.

Reply:
Forgot to mention where to exactly use the "default-ness"? Possibly within `CreatorCollector.verifyNonDup()` when `CreatorCollector.addStringCreator()` is called.

Reply:
Quick note: I think the term should be "default creator" but in some places went with "Primary Creator", like in `POJOPropertiesCollector._addCreators()`. But I will change that now.

The default constructor of the Record class is absolutely not explicit one since there is no annotation. So if there are true explicit creators, it should be ignored.
Note, too, that POJOs (and Kotlin/Scala equivalents) can introduce Primary creators as well; via `AnnotationIntrospector.

Conversely tho it has higher precedence than implicit ones: so three levels really:

1. Explicit creators -- annotations (or anything `AnnotationIntrospector` indicates)
2. Default Creator
3. Implicit Creators

Reply:
So, basically "default-ness" is not carried along `BasicDeserializerFactory` and it should not need to.
But it is considered during `POJOPropertiesCollector._addCreators()`.

It looks like somehow default constructor is found as-is, but also as explicit one.
This latter part seems odd... I'll see if `PotentialCreator` might be constructed incorrectly or something.

Reply:
One other note: "default creator" (or "default constructor") can be confusing wrt `JsonCreator.Mode` -- `DEFAULT` for that is completely unrelated (rather that `DEFAULT` basically means `mode` is unspecified/undefined).